src/main/java/org/apache/commons/compress/compressors/lzma/LZMACompressorInputStream.java

```diff
@@ -56,7 +56,7 @@ public LZMACompressorInputStream(final InputStream inputStream)
             in = new LZMAInputStream(inputStream, MemoryLimit.getMemoryLimitInKb());
         } catch (org.tukaani.xz.MemoryLimitException e) {
             //convert to commons-compress exception
-            throw new MemoryLimitException("exceeded calculated memory limit", e);
+            throw new MemoryLimitException(e.getMemoryNeeded(), e.getMemoryLimit());
```

Comment:
Not looking further into the source code of both MemoryLimitException classes, couldn't this hide the root cause of the exception? As the original `e` exception is ignored?